plugins/producer_plugin/test/test_options.cpp

```diff
@@ -0,0 +1,57 @@
+#define BOOST_TEST_MODULE full_producer_trxs
+#include <boost/test/included/unit_test.hpp>
+
+#include <eosio/producer_plugin/producer_plugin.hpp>
+
+#include <eosio/testing/tester.hpp>
+
+#include <eosio/chain/genesis_state.hpp>
+#include <eosio/chain/thread_utils.hpp>
+#include <eosio/chain/trace.hpp>
+#include <eosio/chain/name.hpp>
+
+#include <appbase/application.hpp>
+
+using namespace eosio;
+using namespace eosio::chain;
+
+BOOST_AUTO_TEST_SUITE(program_options)
+
+BOOST_AUTO_TEST_CASE(state_dir) {
```

Comment:
Instead of adding a complete new test executable for this. Can we not just add a `--state-dir` option to the `test_trx_full` test and verify it is honored there. Seems like a large overhead for such a small test.

Reply:
I think we eventually really need to reorganize a lot of our tests (though not for this PR). For example, some of these tests (and perhaps the test in this PR is also the case) are really trying to bring up a minimal Leap environment (with appbase and plugins) for testing but without having to resort to a Python integration test that actually leverages the Leap binaries. Those kinds of tests don't seem to belong as unit tests of any particular plugin (often these tests will need to bring up the minimum required plugins of the chain plugin and producer plugin), so I am not sure why we should put the tests in the directory of one of the plugins.

Even for tests isolated to unit testing the functionality of the just one plugin, why do we need multiple executables rather than combining them together into one executable as we do with our other unit tests.

Anyway, I'm not suggesting such major reorganization to be done as part of this PR. But to me it seems like what we really should do is move this test to the chain_plugin directory for now. Then later either integrate those chain_plugin tests together into a collection of test suites built into one executable (similar to the unit_test executable), or if appropriate move them out of the plugins and put them somewhere else entirely (again probably as one executable).

What are your thoughts on that approach @heifner?

Reply:
Discussed in our meeting.

The current issue with these minimal Leap environment (with appbase and plugins) is that appbase does not currently support teardown/startup which is why there is only one test per executable.

For this PR:
Move Leap env tests that include appbase into `tests/xxx` where `xxx` name is up for suggestions.

Future work:
- Enhance appbase to allow for teardown/startup. Currently you can only run one test per executable that does a plugin startup/shutdown because of appbase limitation.
- Merge existing plugin/tests into one executable to reduce link time and disk usage.

Reply:
I created an issue to track the limitation in appbase that prevents having more than one test per executable: https://github.com/AntelopeIO/leap/issues/585.

We discussed how we cannot just move to the chain_plugin directory, so are options are either keeping it here or moving it into `tests`. Ultimately, it should be moved into `tests`. But we have more things to figure out regarding how we wish to organize our tests and we don't want that to hold up this PR indefinitely. So we agreed that for now it can exist in the producer_plugin directory and will need to be moved later (along with many other tests) as part of that reorganization effort (tracked in https://github.com/AntelopeIO/leap/issues/586).

Reply:
After discussing with Areg, we'll leave it as is for now, but we plan on creating a new directory for `system` level tests, and address the appbase 1 test instance per executable limitation.